lib/graphql/static_validation/rules/variable_usages_are_allowed.rb

```diff
@@ -7,32 +7,35 @@ class VariableUsagesAreAllowed
       def validate(context)
         # holds { name => ast_node } pairs
         declared_variables = {}
-
         context.visitor[GraphQL::Language::Nodes::OperationDefinition] << ->(node, parent) {
           declared_variables = node.variables.each_with_object({}) { |var, memo| memo[var.name] = var }
         }
 
         context.visitor[GraphQL::Language::Nodes::Argument] << ->(node, parent) {
-          return if !node.value.is_a?(GraphQL::Language::Nodes::VariableIdentifier)
-          arguments = nil
-
-          case parent
-          when GraphQL::Language::Nodes::Field
-            arguments = context.field_definition.arguments
-          when GraphQL::Language::Nodes::Directive
-            arguments = context.directive_definition.arguments
-          when GraphQL::Language::Nodes::InputObject
-            arg_type = context.argument_definition.type.unwrap
-            if arg_type.is_a?(GraphQL::InputObjectType)
-              arguments = arg_type.input_fields
+          node_values = Array.wrap(node.value)
+          return if !(node_values.all? { |value| value.is_a? GraphQL::Language::Nodes::VariableIdentifier })
+
+          node_values.each do |node_value|
+            arguments = nil
+
+            case parent
```

Comment:
This case statement is inside the `node_values.each` loop, but does its result depend on `node_value`? 

If we know that the result of this case statement is always the same, can we move it _outside_ the loop? That way, we evaluate it exactly once (rather than evaluating it for each item in `node_values`).